### PR TITLE
WD: Authentication

### PR DESCRIPTION
**Pull request by Prabhav-Sharma**
## Authentication
I've added Authentication to the project, Instead of creating separate pages for login and signup, I've created a single modal, which lets the user navigate between both internally, simplifying and reusing the same variables. 

### Other Changes
1. Add Auth provider
2. Add Util classes for line-clamp and modal background
3. Add API calls for authentication
4. Add custom hooks for toggle and AuthForm
5. Add FormInput component 

### Important files to check 
- [ ] src/Hooks/useAuthForm.jsx
- [ ] src/Hooks/useToggle.jsx
- [ ] src/components/AuthModal.jsx
- [ ] src/components/FormInput.jsx
- [ ] src/contexts/providers/AuthProvider.js
- [ ] src/contexts/reducers/authReducer.js
- [ ] src/services/authAPI.js
- [ ] src/styles/utils.css

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import "./App.css";
-import Navigation from "./components/Navigation/Navigation";
+import { Navigation } from "./components";
```
Comment (biswanath-tewari) [inline src/App.js:2]: Liked the named export method for importing all the components

File: src/Hooks/useAuthForm.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { useReducer } from "react";
```
Comment (biswanath-tewari) [inline src/Hooks/useAuthForm.jsx:1]: this could have been kept in a context file instead of placing it in a custom hook file

**Review by biswanath-tewari — APPROVED** [review]

File: src/Hooks/useAuthForm.jsx (hunk 2 above)
Comment (Prabhav-Sharma) [reply src/Hooks/useAuthForm.jsx:1]: useAuthForm doesn't create a context, it just creates and maintains the state for the Authentication form inputs. My options were either to create a separate file for the reducer in the components folder or to keep the logic in one place for better readability,  so I went with this, as didn't want to create nested folders and this made sense at the time.

**Review by devananth — APPROVED** [review]
- Good Work 
- Liked the way you used *useAuthForm* for both **login** and **signup**

**Pull request merged**